### PR TITLE
Feat/1688 - Create a new type (Grant Document) and two subtypes (Proposal, Report) #1156

### PR DESCRIPTION
**Pull request by niezgoda**
https://trello.com/c/JS4dPCx7/1688-create-a-new-type-grant-document-and-two-subtypes-proposal-report

Requirements
- add the following Type for RO: Grant Document
- add the following Subtypes for RO: Proposal, Report
- make sure we use the Proposal template for Grant Documents
- remove Proposal from Squidex schema but not from our model

Note
- don't remove Proposals search filter until the migration is done

Out of scope
- update the search filter by adding Grant Documents as an option

### CODE REVIEW

File: apps/asap-server/test/controllers/research-outputs.test.ts [hunk 1]
```diff
@@ -454,15 +480,15 @@ describe('ResearchOutputs controller', () => {
       test('Should pass the filter parameter as a squidex filter', async () => {
```
Comment (peterstarling) [inline apps/asap-server/test/controllers/research-outputs.test.ts:480]: i guess we should probably test searching for Proposals and Grant Documents separately - i would expect the first one to use just the Proposal filter and the other one - both?

File: apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx [hunk 2]
```diff
@@ -69,27 +69,27 @@ const renderComponent = async () => {
   return result;
 };
 
-describe('a proposal research output', () => {
+describe('a Grant Document research output', () => {
   it('renders with its teams', async () => {
```
Comment (peterstarling) [inline apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:73]: i suppose this test should show the difference in rendering proposals or grant documents and other types - i think for the time being you should show that both Grant Documents and Proposal types are rendered differently (i think the test technically show that only the grant document one is different)

File: apps/frontend/src/shared-research/__tests__/api.test.ts [hunk 3]
```diff
@@ -84,12 +84,14 @@ describe('getResearchOutputs', () => {
   it('builds a multiple filter query', async () => {
     await getResearchOutputs(mockIndex, {
       ...options,
-      filters: new Set<ResearchOutputType>(['Article', 'Proposal']),
+      filters: new Set<ResearchOutputType>(['Article', 'Grant Document']),
```
Comment (peterstarling) [inline apps/frontend/src/shared-research/__tests__/api.test.ts:87]: so if the Proposal is still being left should we not keep this test and add a new one to show how grant documents are searched?

File: apps/storybook/src/SearchAndFilter.stories.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const Normal = () => (
     searchQuery={text('Search Query', '')}
     filterTitle={text('Filter Title', 'TYPE OF OUTPUTS')}
     filterOptions={[
-      { label: 'Proposal', value: '1' },
+      { label: 'Grant Documentdata/type/iv', value: '1' },
```
Comment (peterstarling) [inline apps/storybook/src/SearchAndFilter.stories.tsx:18]: why the `/type/iv`?

File: apps/frontend/src/shared-research/api.ts [hunk 5]
```diff
@@ -35,7 +35,8 @@ export const researchOutputTypeFilters: Record<
   ResearchOutputType,
   { filter: string }
 > = {
-  Proposal: { filter: 'type:Proposal' },
+  'Grant Document': { filter: 'type:Proposal OR type:"Grant Document"' },
+  Proposal: { filter: 'type:Proposal OR type:"Grant Document"' },
```
Comment (peterstarling) [inline apps/frontend/src/shared-research/api.ts:39]: so you are leaving Proposal in the api yeah?

**Review by peterstarling — CHANGES_REQUESTED** [review]

**Comment by peterstarling** [comment]
im not sure if i understood this correctly but i think you are leaving proposal in the filter and in the parameters and types so i think all the tests for the proposals should remain untouched

File: apps/frontend/src/shared-research/__tests__/api.test.ts (hunk 3 above)
Comment (niezgoda) [reply apps/frontend/src/shared-research/__tests__/api.test.ts:87]: You cannot search by the Proposal. It's removed on the FE
image

File: apps/storybook/src/SearchAndFilter.stories.tsx (hunk 4 above)
Comment (niezgoda) [reply apps/storybook/src/SearchAndFilter.stories.tsx:18]: whoops.

File: apps/frontend/src/shared-research/api.ts (hunk 5 above)
Comment (niezgoda) [reply apps/frontend/src/shared-research/api.ts:39]: That's true

File: apps/asap-server/src/entities/research-output.ts [hunk 6]
```diff
@@ -133,6 +136,7 @@ const isSharingStatus = (
   (sharingStatuses as ReadonlyArray<string>).includes(status);
 
 const isResearchOutputType = (type: string): type is ResearchOutputType =>
+  type !== 'Proposal' &&
```
Comment (kmaid) [inline apps/asap-server/src/entities/research-output.ts:139]: This seems strange. Why are proposals not a research output type? If proposals are remapped and never exposed to the frontend can we remove this type from the modal?

File: packages/react-components/src/templates/SharedResearchGrantDocument.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
+import { css } from '@emotion/react';
```
Comment (kmaid) [inline packages/react-components/src/templates/SharedResearchGrantDocument.tsx:1]: I don't think this file exists anymore after my PR

**Review by kmaid — CHANGES_REQUESTED** [review]

**Review by peterstarling — APPROVED** [review]

**Review by kmaid — APPROVED** [review]

**Pull request merged**